app/src/main/java/com/google/samples/apps/nowinandroid/ui/NiaAppState.kt

```diff
@@ -164,3 +168,23 @@ private fun NavigationTrackingSideEffect(navController: NavHostController) {
         }
     }
 }
+
+internal fun checkAppThemeOnLaunch(uiModeManager: UiModeManager, darkTheme: Boolean) {
```

Comment:
@tunjid @alexvanyo IMO, this function shouldn't be here because it's not part of the `NiaAppState` API surface. Also, `NiaAppState` responsibility is to handle the UI logic of the `NiaApp` composable function, not the `MainActivity`. 

I'd probably have this as a private function in the Activity itself, and would call it `setAppTheme` instead of `checkAppThemeOnLaunch`. It's not checking anything, it's setting the theme, and also, if this is called "on launch" or not is something this function shouldn't care about